tests/operators/test_Fourier.py

```diff
@@ -1,42 +1,96 @@
-# %%
+import pytest
 import torch
 
 from mrpro.data import KTrajectory
 from mrpro.data import SpatialDimension
 from mrpro.operators import Fourier
 
-Nb, Nc, Nz, Ny, Nx = 8, 12, 1, 320, 320
-image = torch.rand(Nb, Nc, Nz, Ny, Nx) + 1j * torch.rand(Nb, Nc, Nz, Ny, Nx)
 
-kx = torch.rand(Nb, 1, Ny, Nx)
-ky = torch.rand(Nb, 1, Ny, Nx)
-kz = torch.rand(1, 1, 1, 1)
+@pytest.mark.parametrize(
+    'im_shape, nkx, nky, nkz',
+    [
+        # 3d rad mri, 4 coils, 1 batches
+        (
+            (1, 4, 8, 16, 32),  # im shape
+            (1, 16, 32, 64),  # kx
+            (1, 16, 32, 64),  # ky
+            (1, 16, 32, 64),  # kz
+        ),
+        # 2d rad cine mri with 8 cardiac phases, 12 coils
+        (
+            (8, 12, 1, 320, 320),
+            (8, 1, 640, 640),
+            (8, 1, 640, 640),
+            (8, 1, 1, 1),
+        ),
+    ],
+)
+def test_fourier(im_shape, nkx, nky, nkz):
+    """Test forward and adjoint applicatio of Fourier Operator."""
 
-ktraj = KTrajectory(kz, ky, kx)
+    # generate random image
+    image = torch.rand(*im_shape) + 1j * torch.rand(*im_shape)
 
-im_shape = SpatialDimension(Nz, Ny, Nx)
-op = Fourier(im_shape=im_shape, traj=ktraj, norm='ortho')
+    # create random trajectories
+    kx = torch.rand(*nkx)
+    ky = torch.rand(*nky)
+    kz = torch.rand(*nkz)
+    ktraj = KTrajectory(kz, ky, kx)
 
-# %%
-with torch.no_grad():
-    kdata = op(image)
-    reco = op.H(kdata)
+    # create operator
+    im_shape = SpatialDimension(im_shape[-3], im_shape[-2], im_shape[-1])
+    op = Fourier(im_shape=im_shape, traj=ktraj, norm='ortho')
 
+    # apply forward and adjoint operator
+    with torch.no_grad():
```

Comment:
torch.no_grad is not needed here, as non of the inputs is requiere_grads anyway.